aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -1023,6 +936,8 @@ bool TensorIterator::fast_set_up(const TensorIteratorConfig& config) {
             TORCH_INTERNAL_ASSERT(op.is_type_defined(), "no type for operand", i);
             op.tensor = at::empty(shape_, op.options(), MemoryFormat::Contiguous);
             op.current_dtype = op.target_dtype;
+          } else if (op.will_resize) {
```

Comment:
Do we have an invariant that it will never be the case that a tensor is undefined AND `will_resize` is true?

Reply:
Yes, `will_resize` can be set to `true` only in `mark_resize_outputs`, and `tensor.defined()` is a necessary condition to enter that branch.